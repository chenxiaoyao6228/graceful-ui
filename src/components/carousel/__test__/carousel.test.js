import { render } from '@testing-library/vue';
import Carousel from '../carousel.vue';
import CarouselItem from '../carousel-item.vue';

describe('Carousel', () => {
  const testComponent = {
    components: { Carousel, CarouselItem },
    template: `
        <carousel v-model="value">
            <carousel-item v-for="item in list" :key="item">
                <div class="box">{{item}}</div>
            </carousel-item>
        </carousel>
      `,
    data() {
      return {
        value: 2,
        list: [0, 1, 2, 3]
      };
    }
  };
  test('match snapshots', () => {
    const { container } = render(testComponent);
    expect(container).toMatchSnapshot();
  });
});
