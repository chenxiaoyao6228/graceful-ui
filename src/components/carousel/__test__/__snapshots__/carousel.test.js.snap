// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel match snapshots 1`] = `
<div>
  <div
    class="g-carousel"
  >
    <div
      class="g-carousel-window"
    >
      <div
        class="g-carousel-container"
      >
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
    </div>
     
    <div
      class="g-carousel-dots"
    />
     
    <!---->
     
    <!---->
  </div>
</div>
`;
